search-spec/openapi.yml

```diff
@@ -0,0 +1,229 @@
+          $ref: '#/components/schemas/Temporal'
+        taskingProperties:
+          $ref: '#/components/schemas/TaskingProperties'
+        productConfiguration:
+          $ref: '#/components/schemas/ProductConfiguration'
+        priority:
+          $ref: '#/components/schemas/Priority'
+        price:
+          $ref: '#/components/schemas/Price'
+        lifecycle:
+          type: array
+          example: [ "FEASIBILITY", "PRICE", "DELIVERY" ]
+          items:
+            type: string
+    Temporal:
+      type: object
+      required:
+        - interval
+      properties:
+        interval:
+          type: array
+          minItems: 1
+          items:
+            type: array
+            minItems: 2
+            maxItems: 2
+            items:
+              type: string
+              nullable: true
+              format: date-time
+              pattern: (\\+00:00|Z)$
+    Spatial:
+      type: object
+      required:
+        - geometry
+      properties:
+        geometry:
+          $ref: https://geojson.org/schema/Geometry.json
+    TaskingProperties: # this is everything that's in slotcollection and affects number of slots
+      type: object
+      description: "Satellite properties that affect number of slots. 
+      Every property of type array creates slotcollection for every combination with other fields. Every property of type range doesn't, i.e. 
+      all existing slot collection need to be within range"
+      properties:
+        provider:
+          type: array
+          items:
+            type: string
+          example: ["Airbus", "Capella"]
+        satelliteType:
+          type: array
+          items:
+            type: string
+          enum: [ "SAR", "Hyperspectral", "Optical", "Aerial" ]
+          example: [ "Optical" ]
+        sensor:
+          type: array
+          items:
+            type: string
+          example: [ "PHR", "SPOT" ]
+        imageMode:
+          type: array
+          items:
+            type: string
+            example: StripMap
+          example: [ "StripMap", "SpotLight" ]
+        incidenceAngle:
+          type: array
+          minItems: 1
+          example: [0, 5]
+          items:
+            type: array
+            minItems: 2
+            maxItems: 2
+            items:
+              type: integer
+              nullable: true
+        cloudCoverage:
+          type: array
+          minItems: 1
+          example: [0, 100]
+          items:
+            type: array
+            minItems: 2
+            maxItems: 2
+            items:
+              type: integer
+              nullable: true
+    ProductConfiguration: # this is everything thats is in slotcollection
+      type: object
+      description: "Values which do not effect the slots availibility"
+      properties:
+        processingLevel:
+          type: string
+        imageFormat:
+          type: string
+        resolution:
+          type: string
+          example: uint16
+    Price:
```

Comment:
That's rather a `PriceRange`, not `Price`.

Reply:
Since this is the request, I'd propose this be a `PriceRange`, which really is just another search constraint dimension.

I'm not sure the circumstances under which a list would be supplied.  Perhaps if we allowed different units?  One could ask for something between _$x_ and _$y_ or between _A_ and _B_ _up42Credits_?

Reply:
> I'm not sure the circumstances under which a list would be supplied
Same here tbh.

@DylanBartels ?

Reply:
We decided to use one datatype for all things price related to start a discussion on this exact topic, we were unsure how to advance to a finer granularity. 

We chose a possible range because with the `PriceRequest` one could want a range, and with the `SlotCollection` the price could indicate the price range of the slots inside the collection. I propose to advance and split price in the following:

- `RequestPrice` (a price range)
- `SlotPriceOffer` (integer)
- `SlotCollectionPriceOfferings` (a price range)

Including the possibility of different units. Let me know what you think.